### PR TITLE
Cleanup some more of copy method/constructor

### PR DESCRIPTION
**Pull request by Susucre**
part of #11054

Another group of manual changes.

### CODE REVIEW

**Comment by xenohedron** [comment]
this looks like individual cards that would be covered by the letter-by-letter regex, is there a reason to use this PR?

**Comment by Susucre** [comment]
I need to clean thks PR as some of it is  indeed handled by the letter PR's. (and in a not auto formatted way)
However it does contain some additional change: the copy method was sometimes not declared as returning the class, but instead the one extended.

File: Mage.Sets/src/mage/cards/t/TamiyoTheMoonSage.java [hunk 1]
```diff
@@ -62,7 +63,8 @@ public TamiyoTheMoonSage copy() {
     }
 }
 
-class TappedCreaturesControlledByTargetCount implements DynamicValue {
+enum TappedCreaturesControlledByTargetCount implements DynamicValue {
```
Comment (Susucre) [inline Mage.Sets/src/mage/cards/t/TamiyoTheMoonSage.java:66]: So this was transformed as an enum there, I can remove it and group it with the other class => enum if you really want to.

**Comment by Susucre** [comment]
Alright aside from Tamiyo, all the cards had at least a copy method not returning the exact class.
All that was done manually, sorry the auto formatter was enabled.

**Review by xenohedron — APPROVED** [review]

**Pull request merged**